### DISCUSSION TITLE
List streams api works but start recording api gives could not find application

### DISCUSSION THREAD

**Discussion by pankajjaiswal-capanicus** [OvenMediaLabs/OvenMediaEngine#1520] [Help]
Hi there,

I have managed to enable the API and it seems to work for get streams.

Here is a request for get streams:
curl -H 'Authorization: Basic xxx https://domain:8443/v1/vhosts/default/apps/app/streams

And here is the response:
{"message":"OK","response":["aaa"],"statusCode":200}

Now, when I try to send request for start recording, I get could not find application:

curl -X POST -H 'Authorization: Basic xxx' -H 'Content-Type: application/json' -d '{"id":"aaa-id-1","stream":{"name": "aaa"}}' https://domain:8443/v1/vhosts/default/apps/app:startRecord

Please suggest what's I'm doing wrong since the list streams api clearly picks up the vhost and the application but recording stream api doesn't picks up the same.

**Comment by bchah** [comment]
Can we see the OME log and your Server.xml to confirm the configuration?

**Comment by getroot** [comment]
I think you probably don't have File Publisher configured in your application.
https://airensoft.gitbook.io/ovenmediaengine/recording-experiment#file-publisher

**Comment by pankajjaiswal-capanicus** [comment]
@getroot works after configuring File Publisher, recording is working now, thanks.
